core/src/Kokkos_Serial.hpp

```diff
@@ -97,7 +97,9 @@ class SerialInternal {
                                size_t thread_local_bytes);
 
   HostThreadTeamData m_thread_team_data;
-  bool m_is_initialized = false;
+  bool m_is_initialized  = false;
+  bool m_was_initialized = false;
```

Comment:
Why do we need both `m_is_initialized` and `m_was_initialized`?

Reply:
because there is a difference between having been initialized preivously, but then being finalized again, and being initialized right now.

Reply:
Does the invariance `m_is_initialized == m_was_initialized && !m_was_finalized` hold?